docs/quiche_integration.md

```diff
@@ -0,0 +1,33 @@
+### Overview
```

Comment:
Excellent doc!

Reply:
nit: can we move to https://github.com/envoyproxy/envoy/tree/master/source/docs? This is where we currently have our dev docs. The top level docs directory is for user docs.